core/core.go

```diff
@@ -29,20 +29,22 @@ import (
 )
 
 const (
-	c_maxAppendQueue                    = 1000000 // Maximum number of future headers we can store in cache
-	c_maxFutureTime                     = 30      // Max time into the future (in seconds) we will accept a block
-	c_appendQueueRetryPeriod            = 1       // Time (in seconds) before retrying to append from AppendQueue
-	c_appendQueueThreshold              = 1000    // Number of blocks to load from the disk to ram on every proc of append queue
-	c_processingCache                   = 10      // Number of block hashes held to prevent multi simultaneous appends on a single block hash
-	c_primeRetryThreshold               = 1800    // Number of times a block is retry to be appended before eviction from append queue in Prime
-	c_regionRetryThreshold              = 1200    // Number of times a block is retry to be appended before eviction from append queue in Region
-	c_zoneRetryThreshold                = 600     // Number of times a block is retry to be appended before eviction from append queue in Zone
-	c_maxFutureBlocks                   = 100     // Number of blocks ahead of the current block to be put in the hashNumberList
-	c_appendQueueRetryPriorityThreshold = 5       // If retry counter for a block is less than this number,  then its put in the special list that is tried first to be appended
-	c_appendQueueRemoveThreshold        = 10      // Number of blocks behind the block should be from the current header to be eligble for removal from the append queue
-	c_normalListProcCounter             = 5       // Ratio of Number of times the PriorityList is serviced over the NormalList
-	c_statsPrintPeriod                  = 60      // Time between stats prints
-	c_appendQueuePrintSize              = 10
+	c_maxAppendQueue                           = 1000000 // Maximum number of future headers we can store in cache
+	c_maxFutureTime                            = 30      // Max time into the future (in seconds) we will accept a block
+	c_appendQueueRetryPeriod                   = 1       // Time (in seconds) before retrying to append from AppendQueue
+	c_appendQueueThreshold                     = 1000    // Number of blocks to load from the disk to ram on every proc of append queue
+	c_processingCache                          = 10      // Number of block hashes held to prevent multi simultaneous appends on a single block hash
+	c_primeRetryThreshold                      = 1800    // Number of times a block is retry to be appended before eviction from append queue in Prime
+	c_regionRetryThreshold                     = 1200    // Number of times a block is retry to be appended before eviction from append queue in Region
+	c_zoneRetryThreshold                       = 600     // Number of times a block is retry to be appended before eviction from append queue in Zone
+	c_maxFutureBlocks                   uint64 = 3       // Number of blocks ahead of the current block to be put in the hashNumberList
```

Comment:
What major difference does 3 or 100 make on maxFutureBlocks? I believe you mentioned this impacts append times. Are we just gauging it off of CPU usage?

Reply:
I see we overwrite it below for the Zone / Region. How are we deciding these numbers?

Reply:
This is how many prime blocks ahead we will try to append at the same time

Reply:
comment added